docs/_docs/04_reference/02-cli-options.md

```
@@ -2450,6 +2451,25 @@
+**Environment Variable**: `TG_FAIL_FAST` (set to `true`)<br/>
+
+When enabled, Terragrunt will fail the entire run as soon as any unit fails. This means that if any unit encounters an error during execution, all remaining units will be skipped and the run will exit with a failure status immediately.
+This is useful for enforcing strict failure handling in CI/CD pipelines or when you want to stop processing on the first error.
+
+Example:
+
+```bash
+terragrunt run --all apply --fail-fast
+```
+
+NOTE: Currently, `--fail-fast` is honored only when the experimental *runner-pool* execution engine is enabled (`--experiment runner-pool` or the matching env var).
```

markdownlint-cli2 (0.17.2)

Emphasis style
Expected: underscore; Actual: asterisk

(MD049, emphasis-style)